druid/src/text/editable_text.rs

```diff
@@ -58,6 +58,12 @@ pub trait EditableText: Sized {
     /// Get the next codepoint offset from the given offset, if it exists.
     fn next_codepoint_offset(&self, offset: usize) -> Option<usize>;
 
+    /// Get the line beginning offset from the given offset, if it exists.
+    fn line_beginning(&self, offset: usize) -> Option<usize>;
+
+    /// Get the line ending offset from the given offset, if it exists.
+    fn line_ending(&self, offset: usize) -> Option<usize>;
+
```

Comment:
Let's call these `left_end_of_line` and `right_end_of_line`. 'beginning' and 'end' are ambiguous, since in right-to-left languages the beginning of the line is on the right. :)